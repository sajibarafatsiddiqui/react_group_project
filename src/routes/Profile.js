import ReservedRockets from 'components/ReservedRockets';
import ReservedMissions from '../components/ReservedMissions';

const Profile = () => (
  <div className="d-flex p-5 gap-5">
    <ReservedMissions />
    <ReservedRockets />
  </div>
);

export default Profile;
