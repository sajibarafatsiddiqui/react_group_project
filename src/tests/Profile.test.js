/* eslint-disable import/no-extraneous-dependencies */
import { configureStore, createSlice } from '@reduxjs/toolkit';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import Profile from '../routes/Profile';

describe('Tests for <Profile /> component', () => {
  const missionsSlice = createSlice({
    name: 'missions',
    initialState: {
      missions: [{
        missionId: '1',
        missionName: 'Mission 1',
        description: 'Mission desc 1',
        reserved: false,
      }, {
        missionId: '2',
        missionName: 'Mission 2',
        description: 'Mission desc 2',
        reserved: false,
      }],
      isLoading: false,
    },
  });

  const rocketsSlice = createSlice({
    name: 'rockets',
    initialState: {
      rockets: [{
        id: '1',
        name: 'rocket 1',
        images: ['1.jpg'],
        description: 'rocket desc 1',
      }, {
        id: '2',
        name: 'Rocket 2',
        images: ['2.jpg'],
        description: 'rocket desc 2',
        reserved: false,
      }],
      ifSucceed: true,
      isLoading: false,
    },
  });

  const store = configureStore({
    reducer: {
      rockets: rocketsSlice.reducer,
      missions: missionsSlice.reducer,
    },
  });

  const wrapper = (component) => <Provider store={store}>{component}</Provider>;

  it('Should render messages when there are no reserved missions and rockets', () => {
    render(wrapper(<Profile />));
    expect(screen.getByText('You have not reserved any missions yet.')).not.toBeNull();
    expect(screen.getByText('You have not reserved any rockets yet.')).not.toBeNull();
  });
  it('Should match the snapshot', () => {
    expect(renderer.create(wrapper(<Profile />)).toJSON()).toMatchSnapshot();
  });
});
