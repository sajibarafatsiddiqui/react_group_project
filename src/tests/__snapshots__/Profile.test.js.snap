// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for <Profile /> component Should match the snapshot 1`] = `
<div
  className="d-flex p-5 gap-5"
>
  <div>
    <h3>
      My Missions
    </h3>
    <p>
      You have not reserved any missions yet.
    </p>
  </div>
  <div>
    <h3>
      My Rockets
    </h3>
    <p>
      You have not reserved any rockets yet.
    </p>
  </div>
</div>
`;
